crypto/pkcs7/pk7_attr.c

```diff
@@ -98,12 +98,18 @@ int PKCS7_add_attrib_content_type(PKCS7_SIGNER_INFO *si, ASN1_OBJECT *coid)
 
 int PKCS7_add0_attrib_signing_time(PKCS7_SIGNER_INFO *si, ASN1_TIME *t)
 {
-    if (t == NULL && (t = X509_gmtime_adj(NULL, 0)) == NULL) {
+    ASN1_TIME *tmp = NULL;
```

Comment:
Do we need the new stack variable here?  If PKCS7_add_signed_attribute fails, can't we just call ASN1_STRING_free(t) instead?